### ISSUE TITLE
Old forum going down

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/cslaforum#383]
Hello everyone,

I am sad to say that the old archived forum (forum.lhotka.net) needs to go away ASAP. It is running on a 32 bit Windows 2003 server and so is vulnerable to this latest virus running around the world. And it can't be upgraded because it is 32 bit.

My current plan is to dump all the data out of the SQL Server db and shut down the VM.

Exactly what to do with the data from the database is unknown at this point, but at least I'll still have a copy in case someone wants to figure out a way to put it back online in some form.

**Comment by ajj7060** [comment]
@rockfordlhotka  Is it possible to send a request to archive.org to crawl it before it goes away?

**Comment by rockfordlhotka** [comment]
I'm not sure how to do that - just send them an email?

**Comment by ajj7060** [comment]
@rockfordlhotka  Hmm, it may have already found it in the past.  Looks like there are snapshots but who knows how many threads it found :-)

**Comment by rockfordlhotka** [comment]
fwiw, I was able to get all the data from the database. So if someone did want to write a simple web site to allow searching through the archived content I'm happy to host such a site - I just don't have time to write it.

**Comment by davimack** [comment]
What format is the data in now (ie, a .bak or other), what version mssql, any private data needs sanitized (eg saved passwords or hashes)? What format would you like the output in (eg, would dumping to static HTML make you happy)? Is it upgradable to a modern flavor of the same engine?

**Comment by rockfordlhotka** [comment]
To my knowledge, the Community Server forum software no longer exists as a product. Even if it does - we're talking about several years of updates that'd need to be applied to get it running again - and I'm not sure the VM image still exists. In any case, I'm not interested in running a whole VM for archive data access.

What I did was dump the data from the SQL Server to another SQL Server (2016 I think) on a different server - so it is basically a temporary backup. From there we can do whatever - but there's a _ton_ of irrelevant information in there beyond the posts and replies, and none of that needs to stick around (imo).

What I have in mind is relatively straightforward - just a simple site that perhaps lists all the top level posts, and when you click on one it gives you the threaded replies. Along with search capability - or perhaps just rely on Google to index it - either way, searching is important obviously.

I think it is around 60k posts/replies. So the number isn't high from a db perspective, but it is high from the thought of creating one massive html doc - I can't see that working well.

**Comment by davimack** [comment]
Ahh. Threaded forum discussions.  I was thinking more along the lines of larger posts, which could be dumped out as individual pages with associated links, indexed by whomever, and just stashed on a server capable of serving out html.

I wonder if we could shoe-horn the data into something like FudForum, or massage it enough to fit into whatever forum software you're currently using (if you are).  Of course, FudForum would require a server to host it, but it runs fine alongside other things - it's likely more than you're interested in maintaining, though.

To summarize: you don't want to have to maintain a separate VM for this, but presumably you're comfortable maintaining a site which queries a database & are comfortable maintaining that database.

So, the problem is more along the lines of finding a migration path to a software which meets your needs.  The migration path can be hacked together, since it just needs to run once (although correctly), which is made more difficult because we do not have an easy way to verify that the results match (since the original site is down?).

**Comment by rockfordlhotka** [comment]
That pretty much sums it up, yes.

If we can get it into a set of linked html files (static content) then I can just host it here on GitHub as a repo. Hadn't really thought about that, but such a thing would certainly work.

The forum was replaced by this cslaforum GitHub repo we're using to converse right now. I decided there was no point running any sort of forum software when GitHub provides a very workable solution.

**Comment by davimack** [comment]
I think I'd want to do some basic analysis of the data before I thought it would be OK to dump it out to html - for example, this issue that we're chatting on right now: if this is what the conversations kinda look like, great, but how many of these pages would there be, and could we get it into anything that would resemble something you'd find useful?

I think static, bare-bones html pages, referencing a stylesheet or what have you, would be a good way to go, provided we generated them smartly - so there'd need to be a page for each user, linking out to the various issues upon which they'd commented; there'd need to be the individual issue pages, with appropriate hyperlink anchors to make bookmarking nice (permalinks, basically); there'd need to be some means of scrounging through the pages, which could be accomplished with a Google search widget or similar.  Yeah - seems reasonable, actually.

If we're going to go there, I'd want to do a proof of concept, but it doesn't seem like rocket science, to be honest.  If you give me a database schema I'll see about writing some c# to dump files out, tinker about with that a bit, and see what comes of it.  Not sure when I could get to it, but it doesn't sound high priority, really, nor does it sound high complexity, so it's possible.  I'm moving house & changing jobs within a couple weeks, though, so I'm a bit pinched for time.

**Comment by rockfordlhotka** [comment]
I needed something light to work on today, so I decided to write a quick app to dump the archive posts into a series of (nearly 12k) static html files - one for each thread from the old forum.

This can be regenerated if necessary, but let's see how it goes to start. The old forum contents are now here:

http://cslanet.com/old-forum/

**Comment by davimack** [comment]
And there you go.  A few indexing pages maybe (e.g., a bunch of author pages linking out to the individual posts) and you'd have some rudimentary navigation by poster or something.  But, hey - the content's not lost, and search engines can index.

Sorry I didn't get 'round to doing anything for you here, but glad you haven't lost the content!

**Comment by rockfordlhotka** [comment]
Feel free to enhance the app and submit a PR for it 😄 

I'll happily re-run it to update the pages.

**Comment by jonnybee** [comment]
We should also have a link to the old forum on the home page http://cslanet.com

**Comment by davimack** [comment]
Hey - is there an Author ID field in the database schema?  Am thinking it'd be nice to build up author pages along the way, with links to everything that author has written, maybe, and to link out to the author from the individual post pages.  Not sure it's all that useful - just some more information that might be worth keeping.

**Comment by rockfordlhotka** [comment]
@jonnybee I added a link on the home page as you suggested.

**Comment by rockfordlhotka** [comment]
@davimack yes, there are author name and author id columns, and there's probably one or more tables with profile info from the author - I didn't actually look beyond just the table with the posts.
